backend/routes/auth.js

```diff
@@ -0,0 +1,39 @@
+const express = require("express");
+app.use(express.json());
+const app = express();
+app.use(express.json());
+router = express.Router();
+const argon2 = require("argon2");
+
+router.post("/signup", async (req, res) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
```

Comment:
You should have `null` checks on the client side as well and prevent submitting an incorrect form, but it's better to never trust the client side 🙂
I'd probably show some error messages like min password length even before sending the server request, during the frontend validation.